### PR TITLE
Enforce control deadlines and audit logical Docker healthchecks

### LINKED ISSUE #226: Parse multiline Docker HEALTHCHECK in CI configuration audit
## Problem

`scripts/check-ci-config.sh` reports that the production Dockerfile has no `HEALTHCHECK`, even though the instruction is present and continued across lines. The check approximates Dockerfile syntax one physical line at a time, so a valid multiline instruction produces a false warning. False positives make the audit easier to ignore and can hide a future real regression.

This was identified during the session 067 CI/dependency audit and is intentionally outside the relay hot-path patch.

## Scope

- Reproduce the current warning against the checked-in Dockerfile.
- Parse or normalize Dockerfile continuation lines before evaluating the `HEALTHCHECK` instruction; do not rely on a raw substring that comments can satisfy.
- Preserve detection of `HEALTHCHECK NONE` and a genuinely missing instruction.
- Add fixtures/tests for single-line healthcheck, multiline healthcheck, commented-out text, `HEALTHCHECK NONE`, malformed continuation, and absent healthcheck.
- Keep the script's human and CI output modes stable unless a change is explicitly documented.

## Acceptance criteria

- `scripts/check-ci-config.sh` reports the checked-in production Dockerfile's live multiline healthcheck correctly.
- Removing or commenting out that instruction makes the check fail or emit the intended blocking diagnostic.
- `HEALTHCHECK NONE` is not treated as an active probe.
- Focused script tests and the full CI-config policy suite pass.
- The implementation does not execute Dockerfile content or introduce shell word-splitting/evaluation hazards.

### PR DESCRIPTION
**Pull request by wallstop**
## Summary

- Enforce strict absolute delivery deadlines for all control-capacity reservation paths, including initial transitions and predicate-filtered delivery.
- Preserve already-terminal channel-closed and classified-generation-canceled outcomes at the deadline while preventing late delivery to otherwise-live recipients.
- Audit Docker `HEALTHCHECK` as a logical instruction, including continuations, comments, final-stage selection, malformed forms, heredocs, and port consistency.
- Add paused-time runtime matrices and black-box CI-policy fixtures, correct the prior session’s reviewed-head record, and document the follow-up deadline audit in #233.

## Root causes

Tokio timeout/select behavior could let an inner capacity future win at or after the documented deadline, and the existing Docker policy inspected individual physical lines rather than the assembled logical `HEALTHCHECK` instruction.

## Red evidence

Before the runtime fix, the initial-transition path could report `Delivered` after its one-second deadline when capacity returned at the boundary. Before the policy fix, the checked-in continued `HEALTHCHECK` was reported as missing.

## Validation

- `cargo fmt --all -- --check`
- `cargo clippy --locked --all-targets --all-features -- -D warnings`
- `cargo test --locked --all-features`
- `cargo deny check`
- `bash -n scripts/check-ci-config.sh`
- `shellcheck scripts/check-ci-config.sh`
- Repository documentation, MSRV, workflow-hygiene, markdown, link-text, hook-readiness, pre-commit, and pre-push policy checks
- Three independent adversarial review tracks completed with zero remaining revisions

Closes #226
Refs #205
Refs #233

---

> [!NOTE]
> **Medium Risk**
> Changes observable disconnect semantics on control-path backpressure (slow consumer vs canceled/closed) and tightens CI Dockerfile validation; runtime paths are heavily regression-tested but affect live WebSocket delivery behavior.
> 
> **Overview**
> Closes a **deadline race** on control-message queue waits: capacity that returns at or after the slow-consumer grace can no longer enqueue after expiry. **Initial room-join/reconnect transitions**, **conditional single-recipient delivery**, and **conditional reservations** (lifecycle broadcasts, session plans, replans, room transactions) now use **timer-first biased `select!`** with an absolute deadline, matching the P24 relay path. At expiry, **channel closed** and **classified generation canceled** still win over **`4002 slow_consumer`**; otherwise-live recipients get the documented slow-consumer disconnect.
> 
> **CI Docker policy** (`scripts/check-ci-config.sh`) now parses **logical `HEALTHCHECK` instructions** (backslash continuations, CRLF, comments, final-stage-only `FROM` reset) and **fails closed** on heredoc `RUN`/`COPY`/`ADD` (including `ONBUILD`) that could spoof probes. Only the supported **`curl` localhost `/v2/health`** probe is accepted, with **port matched to `EXPOSE`**.
> 
> Adds **paused-time regression matrices** for the three wait primitives (legacy/classified queues, exact/post-deadline boundaries) and **black-box healthcheck fixture tests**; changelog and session progress docs updated.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d30affde6ae966132c561d71c5887a5f7f8113d4. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by wallstop** [comment]
bugbot run

**Comment by wallstop** [comment]
@copilot review

**Comment by wallstop** [comment]
@cursor review

**Comment by wallstop** [comment]
bugbot run

**Comment by wallstop** [comment]
@copilot review

File: scripts/check-ci-config.sh [hunk 1]
```diff
@@ -102,20 +102,106 @@ if [[ -f Dockerfile ]]; then
...
+                sub(/^[[:space:]]+/, "", instruction)
+                sub(/[[:space:]]+$/, "", instruction)
+                lower = tolower(instruction)
+                if (lower ~ /^from([[:space:]]|$)/) {
+                    healthcheck = ""
+                    return
+                }
+                if (lower ~ /^(run|copy)([[:space:]]|$)/ && instruction ~ /<</) {
+                    audit_error = "unsupported-heredoc"
+                    return
```
[the remark this answers is not in the document]
Comment (wallstop) [reply scripts/check-ci-config.sh:122]: Accepted and fixed in `2fd062e`. The fail-closed matcher now covers direct `RUN`/`COPY`/`ADD` heredocs and their `ONBUILD` wrappers. A compact opener matrix exercises `RUN`, `COPY`, `ADD`, and `ONBUILD ADD` with heredoc bodies that impersonate both `FROM` and `HEALTHCHECK`; all must return the existing blocking `unsupported-heredoc` diagnostic. Focused tests, strict clippy, Bash syntax, ShellCheck, the checked-in audit, and an independent adversarial rereview are green.

**Comment by wallstop** [comment]
bugbot run

**Comment by wallstop** [comment]
@copilot review

**Comment by wallstop** [comment]
bugbot run

**Comment by wallstop** [comment]
@copilot review

**Comment by wallstop** [comment]
Final exact-head audit for `d30affde6ae966132c561d71c5887a5f7f8113d4`:

- All 11 applicable hosted workflows succeeded: Advanced Safety, Browser Interop, CI, Documentation Validation, Fortress Interop, Fortress WASM Interop, Link Check, Markdownlint, Spellcheck, Unused Dependencies, and WebRTC Interop. Dependabot auto-merge was intentionally skipped.
- Cursor Bugbot reviewed this exact head and reported no new issues. Its earlier valid `ADD` heredoc finding was fixed in `2fd062e`; the inline thread is resolved, with zero unresolved review threads remaining.
- Copilot review was retriggered after the final push through both the review API and `@copilot review`; GitHub reported the repository's Copilot review quota is exhausted. No distinct human reviewer is available among repository contributors (the sole human contributor is the PR author).
- Local `HEAD` and the upstream branch are identical at the SHA above, the tracked worktree is clean, and `git diff --check` passes.
- The PR remains mergeable against base `main` at `664f415bd90376237f860ffe65d500de1d1dd536`.
- The first hosted Windows lint failure exposed a platform-specific test-helper signature; that was corrected in `8562708`, and the final Windows lint and nextest jobs are green.

This is the final publication audit; no commit was added after the exact-head review and CI results.
